python-lib/dku_google/gcloud.py

```diff
@@ -35,4 +28,111 @@ def get_access_token_and_expiry(config={}):
     info = json.loads(info_str)
     token_key_chunks = config.get("token-key", "{.credential.access_token}")[2:-1].split('.')
     expiry_key_chunks = config.get("expiry-key", "{.credential.token_expiry}")[2:-1].split('.')
-    return _safe_get_value(info, token_key_chunks), _safe_get_value(info, expiry_key_chunks) 
+    return _safe_get_value(info, token_key_chunks), _safe_get_value(info, expiry_key_chunks)
+
+
+def get_account():
+    account = _safe_get_value(_get_gcloud_info(), ["config", "account"], None)
+    return account
+
+
+def _run_cmd(cmd=None):
+    """
+    Run command via subprocess. Clean retrieval of error message if fails. Trims any trailing space.
+    """
+
+    try:
+        out = subprocess.check_output(cmd).rstrip()
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+    return out
+
+
+def get_project_region_and_zone():
+    """
+    Call the 'gcloud config get-value' command to retrieve parameters.
+    Requires that the gcloud CLI has been properly configured.
+    """
+
+    cmd = ["gcloud", "config", "get-value"]
+    project = _run_cmd(cmd+["core/project"])
+    region = _run_cmd(cmd+["compute/region"])
+    zone = _run_cmd(cmd+["compute/zone"])
+
+    logging.info("The following config params were found for gcloud: PROJECT={}, REGION={}, ZONE={}".format(project, region, zone))
+
+    return project, region, zone
+
+
+def _get_gce_instance_info():
+    """
+    Run 'gcloud compute instances describe <the-host-vm>'
+    Requires the compute.zones.list IAM permission.
+    """
+    
+    gce_instance_info = {}
+
+    project, region, zone = get_project_region_and_zone()
+    logging.info("Retrieving GCE VM info")
+    cmd_base = ["gcloud", "compute", "instances", "describe"]
+    gce_vm_name = socket.gethostname()
```

Comment:
apparently there are better/safer ways to get the instance name, especially using the metadata server. cf https://stackoverflow.com/questions/31688646/get-the-name-or-id-of-the-current-google-compute-instance

Reply:
Refactored a bit the instance info retrieval.